pkg/actor/actor.go

```diff
@@ -72,15 +75,18 @@ func (e ValidationError) Error() string {
 
 // Actor is one action against the cluster if the cluster resource state can be handled
 type Actor interface {
-	Handles([]api.ClusterCondition) bool
 	Act(context.Context, *resource.Cluster) error
 	GetActionType() api.ActionType
 }
 
+type Director interface {
+	GetActionsToExecute(*resource.Cluster) []api.ActionType
+}
+
 // NewOperatorActions creates a slice of actors that control the actions or actors for the operator.
 // The order of the slice is critical so that the actors run in order, for instance update has to
 // happen before deploy.
-func NewOperatorActions(scheme *runtime.Scheme, cl client.Client, config *rest.Config) []Actor {
+func NewOperatorActions(scheme *runtime.Scheme, cl client.Client, config *rest.Config) map[api.ActionType]Actor {
```

Comment:
Rather than exposing this, what do you think about adding this map to `Director` and building it up in a constructor function? Then we can return actors to execute, rather than their types. It would also prevent the need for a reconciler to need actions as well as a director.

E.g.

```golang
type Director interface {
  GetActorToExecute(*resource.Cluster) []Actor
}

func NewDirector(scheme *runtime.Scheme, cl client.Client, config *rest.Config) Director {
  actions := // build map
  return &clusterDirector{scheme: scheme, client: cl, config: config, actions: actions}
}

type clusterDirector struct {
  ...
  actions map[api.ActionType]Actor
}
```

Reply:
Sure, makes sense